docs/docs/modules/agents/agent_types/tool_calling.ipynb

```diff
@@ -0,0 +1,306 @@
+    "\n",
+    "We can take advantage of this structured output, combined with the fact that you can bind multiple tools to a [tool calling chat model](/docs/integrations/chat/) and\n",
+    "allow the model to choose which one to call, to create an agent that repeatedly calls tools and receives results until a query is resolved.\n",
+    "\n",
+    "This is a more generalized version of the [OpenAI tools agent](/docs/modules/agents/agent_types/openai_tools/), which was designed for OpenAI's specific style of\n",
+    "tool calling. It uses LangChain's ToolCall interface to support a wider range of\n",
+    "provider implementations, such as [Anthropic](/docs/integrations/chat/anthropic/), [Google Gemini](/docs/integrations/chat/google_vertex_ai_palm/), and [Mistral](/docs/integrations/chat/mistralai/)\n",
+    "in addition to [OpenAI](/docs/integrations/chat/openai/).\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "Any models that support tool calling can be used in this agent. [TODO ADD WHICH]\n",
+    "\n",
+    "This demo uses [Tavily](https://app.tavily.com), but you can also swap in any other [built-in tool](/docs/integrations/tools) or add [custom tools](/docs/modules/tools/custom_tools/).\n",
+    "You'll need to sign up for an API key and set it as `process.env.TAVILY_API_KEY`.\n",
+    "\n",
+    "```{=mdx}\n",
+    "import ChatModelTabs from \"@theme/ChatModelTabs\";\n",
+    "\n",
+    "<ChatModelTabs customVarName=\"llm\" />\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_anthropic import ChatAnthropic\n",
+    "\n",
+    "llm = ChatAnthropic(model=\"claude-3-sonnet-20240229\", temperature=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initialize Tools\n",
+    "\n",
+    "We will first create a tool that can search the web:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.agents import AgentExecutor, create_tool_calling_agent\n",
+    "from langchain_community.tools.tavily_search import TavilySearchResults\n",
+    "from langchain_core.prompts import ChatPromptTemplate\n",
+    "\n",
+    "tools = [TavilySearchResults(max_results=1)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Agent\n",
+    "\n",
+    "Next, let's initialize our tool calling agent:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt = ChatPromptTemplate.from_messages(\n",
+    "    [\n",
+    "        (\n",
+    "            \"system\",\n",
```

Comment:
This will fail for models that don't accept system messages (I think Vertex). Maybe leave out?